common/api/src/db/peers.rs

```diff
@@ -0,0 +1,216 @@
+        self.last_seen.0 = Some(lower_bound);
+        self
+    }
+
+    pub fn with_last_seen_lte(mut self, upper_bound: SystemTime) -> Self {
+        self.last_seen.1 = Some(upper_bound);
+        self
+    }
+
+    pub fn with_quality_gte(mut self, lower_bound: f64) -> Self {
+        self.quality.0 = Some(lower_bound);
+        self
+    }
+
+    pub fn with_quality_lte(mut self, upper_bound: f64) -> Self {
+        self.quality.1 = Some(upper_bound);
+        self
+    }
+}
+
+/// Status of the peer as recorded by a network component.
+#[derive(Debug, Clone, PartialEq)]
+pub struct PeerStatus {
+    pub id: (OffchainPublicKey, PeerId),
+    pub origin: PeerOrigin,
+    pub last_seen: SystemTime,
+    pub last_seen_latency: Duration,
+    pub heartbeats_sent: u64,
+    pub heartbeats_succeeded: u64,
+    pub backoff: f64,
+    /// Until when the peer should be ignored.
+    pub ignored_until: Option<SystemTime>,
+    pub multiaddresses: Vec<Multiaddr>,
+    // Should be public(crate) but the separation through traits does not allow direct SQL ORM serde
+    pub quality: f64,
+    // Should be public(crate) but the separation through traits does not allow direct SQL ORM serde
+    pub quality_avg: SingleSumSMA<f64>,
+}
+
+impl PeerStatus {
+    pub fn new(id: PeerId, origin: PeerOrigin, backoff: f64, quality_window: u32) -> PeerStatus {
+        PeerStatus {
+            id: (OffchainPublicKey::from_peerid(&id).expect("invalid peer id given"), id),
+            origin,
+            heartbeats_sent: 0,
+            heartbeats_succeeded: 0,
+            last_seen: SystemTime::UNIX_EPOCH,
+            last_seen_latency: Duration::default(),
+            ignored_until: None,
+            backoff,
+            quality: 0.0,
+            quality_avg: SingleSumSMA::new(quality_window as usize),
+            multiaddresses: vec![],
+        }
+    }
+
+    // Update both the immediate last quality and the average windowed quality
+    pub fn update_quality(&mut self, new_value: f64) {
+        debug_assert!(
+            (0.0f64..=1.0f64).contains(&new_value),
+            "quality failed to update with value outside the [0,1] range"
+        );
+
+        if (0.0f64..=1.0f64).contains(&new_value) {
+            self.quality = new_value;
+            self.quality_avg.push(new_value);
+        }
+    }
+
+    /// Gets the average quality of this peer
+    pub fn get_average_quality(&self) -> f64 {
+        self.quality_avg.average().unwrap_or_default()
+    }
+
+    /// Gets the immediate node quality
+    pub fn get_quality(&self) -> f64 {
+        self.quality
+    }
+
+    /// Determines whether the peer is ignored due to quality concerns, given the current time.
+    #[inline]
+    pub fn is_ignored(&self) -> bool {
+        let now = hopr_platform::time::current_time();
+        self.ignored_until
+            .is_some_and(|t| now.saturating_sub(t) == std::time::Duration::from_secs(0))
```

Comment:
```suggestion
            .is_some_and(|t| now <= t)
```

Reply:
Is not this a bug also in our 3.0 codebase too, @Teebor-Choka ?